### ISSUE TITLE
BaseBuilding Sync issue

### ISSUE THREAD

**Issue by dgkepel** [SubnauticaNitrox/Nitrox#397]
Currently running a server succesfully. Events are not always synced but we work around that.

There seem to be an issue with the syncing of building parts. Players seem to have their own instances of certain buildings. Everyone has to deconstruct the same part by themselves. On server restart these parts are back in the save. Tried overwriting. Doing it solo without other players but can't seem to be able to clean it up

**Comment by OmniTitan** [comment]
Have also had this issue, it broke our game as we could not stabilise our structure. At one point we reloaded the server and actually lost a ton of buildings and through this lost most of our items. Hope there is a fix for this as its the only reason we stopped playing.

**Comment by Halokillerman7** [comment]
Yep also have had this issue. With active players always on and getting off building. some of the deleted buildings return on said players screen but not on the others until relog. We have also noticed when placing some buildings it shows 100% but the none building player can not use only get the option to finish construction that they are unavailable to.

**Comment by OmniTitan** [comment]
This is so far the only bug that is actually game breaking as it causes hull integrity issues. Deconstruction needs to be synced like construction. Anything that is built becomes permanent, if a player deconstructs it then on relog it will be back. We tested this and it does allow buildings inside of buildings sort of if you deconstruct one building and then construct a different one in its place. On relog it then has both.

**Comment by Aphermion** [comment]
Yea, I tested with a whole base. Deconstructed it all, then built a new one on top.

As far as I can tell, it doesn't really matter which player built which part of the base.
Some things stayed deleted, some just came back after restarting the server.

We tested deleting and rebuilding while the other player was on the server, 
and tested the same thing while the other player was absent.
Generally the same result every time.
(+ sometimes we just seem to fall out of sync and can't see parts the other player has built, those parts seem to be more likely to come back after a restart even if deleted by the creator)

A picture for your amusement:
![image](https://user-images.githubusercontent.com/31079914/50353630-1edc0d00-0549-11e9-8dc9-4623779298e8.png)
